OpenMage/magento-lts
# New feature: ConfigurableSwatches now allows for auto-generation of the swatch image file based on color selection

### Description (*)

I avoid installing third-party extensions to manage swatches that often need to be integrated into the template and cause issues and mess up the database.
But I also hate explaining to the client how to create and how to upload swatch images.
With this modification, it's possible to choose a color that will be used for dynamically creating the swatch.

The code only comes into play if a fallback image is not present in `media/wysiwyg/swatches/optionvalue.png`.
A resized image is created in cache directly to avoid conflicts with the original images in:
`media/catalog/swatches/STOREID/WidthxHeight/optionvalue.png`.
This way, it is perfectly compatible with the other modes that always take precedence.

![demo](https://github.com/OpenMage/magento-lts/assets/5071467/f451786e-1ddd-40e7-a3f1-a8dec4cd2067)
![demo_picker](https://github.com/OpenMage/magento-lts/assets/5071467/cad78fc0-fe00-46bd-a7b7-b854d0fd1cea)
![demo_delete](https://github.com/OpenMage/magento-lts/assets/5071467/9e06dc0e-92bf-4c1c-9683-43dbb9bbdf7a)


### Manual testing scenarios (*)

1. Configure color to be displayed as a swatch
2. Create a new configurable product with the `color` attribute
3. Create a new simple product with a color option that does not have a fallback image in `media/wysiwyg/swatches`

( If everything is correct, should display a broken image in product view and layered )

4. Navigate to Manage Attributes, edit the color attribute, and select a color for the option with the color picker

## Results:
A new swatch image is created on the fly directly in the cache `media/catalog/swatches/` foreach requested dimensions, both for the product view and layered navigation filters.

### Questions or comments

This feature is heavly inspired to magento 2 and can be extended to allow direct upload of swatch images in
`media/wysywyg/swatches`.
Use `<input type="color" />` for color picker that seems is widely supported in Chrome and Firefox https://caniuse.com/input-color

~~**It is not possible to set a default empty value for input type="color", so all colors are black by default, which is always better than a broken image (for now😉)**~~

I had to update the EAV module by adding a new table `eav_attribute_option_swatch` because it seem to me more consistent with the database schema. If you have any alternative solutions, please let me know.
I have tested also with the `Mage_ConfigurableSwatches` module disabled.

**Please comment only on the PR implementation; for anything else, it's best to open a separate discussion.**

### Contribution checklist (*)
